### DISCUSSION TITLE
Updating BSB-LAN to current version? Useful?

### DISCUSSION THREAD

**Discussion by atl285** [fredlcore/BSB-LAN#592] [General]
Hello. I'm running an old version (0.42) of BSB-LAN on a Arduino MEGA2560 with Ethernet-Shield and early / unknown version of the adapter board:
![bsb_bottom](https://github.com/fredlcore/BSB-LAN/assets/15815417/75abf10e-9147-4ed1-a824-d1c8e2a98974)
![bsb_top](https://github.com/fredlcore/BSB-LAN/assets/15815417/4d2ecb22-eddf-4037-9b7e-108ffa68d0a9)

With this setup I only can read some values (Boiler temperature, boiler return, ionization current) from my Broetje BBS EcoCondens 2N 15 (via OCI420). Now my questions:

1. Make it sense to upgrade BSB-LAN to get more usable values (maybe warm water too) and eventually also be able to write values?
2. What new hardware is needed? Can I used the existing adapter with an ESP32 for example?

**Comment by fredlcore** [comment]
The manual describes how to modify the old adapters like yours in order to work with the ESP32. Note, however, that you will have to connect the BSB-LAN board with the ESP32 with wires then, it won't fit straight on the microcontroller.
I'm not sure if your heater has a LMU64 or LMU74, but in both cases it's not that much likely that you'll get (many) more parameters because the OCI420 only has a very limited set of parameters available anyway. Other features of BSB-LAN might be worth the switch to an ESP32, for example to be able to update over the air etc., but the general BSB functionality is mainly the same.

**Comment by atl285** [comment]
Thanks for the reply. My heater has a LMU54/64. I was hoping to be able to control the timers for start and end of the heating. If the general BSB functionality stays the same, I will continue with the current hardware setup for now.
